x-pack/platform/plugins/shared/streams/server/routes/content_packs/route.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Readable } from 'stream';
+import { z } from '@kbn/zod';
+import {
+  createConcatStream,
+  createListStream,
+  createMapStream,
+  createPromiseFromStreams,
+} from '@kbn/utils';
+import { createSavedObjectsStreamFromNdJson } from '@kbn/core-saved-objects-server-internal/src/routes/utils';
+import { createServerRoute } from '../create_server_route';
+import { StatusError } from '../../lib/streams/errors/status_error';
+
+const downloadContentPacksRoute = createServerRoute({
+  endpoint: 'GET /api/streams/{name}/content_packs',
+  options: {
+    access: 'internal',
+  },
+  params: z.object({
+    path: z.object({
+      name: z.string(),
+    }),
+  }),
+  security: {
+    authz: {
+      enabled: false,
+      reason:
+        'This API delegates security to the currently logged in user and their Elasticsearch permissions.',
+    },
+  },
+  async handler({ params, request, response, getScopedClients, context }) {
+    const { assetClient, soClient, streamsClient } = await getScopedClients({ request });
+
+    await streamsClient.ensureStream(params.path.name);
+
+    const dashboards = await assetClient
+      .getAssets({ entityId: params.path.name, entityType: 'stream' })
+      .then((assets) => assets.filter(({ assetType }) => assetType === 'dashboard'));
+    if (dashboards.length === 0) {
+      throw new StatusError(`No dashboards are linked to [${params.path.name}] stream`, 400);
+    }
+
+    const exporter = (await context.core).savedObjects.getExporter(soClient);
+    const exportStream = await exporter.exportByObjects({
+      request,
+      objects: dashboards.map((dashboard) => ({ id: dashboard.assetId, type: 'dashboard' })),
+      includeReferencesDeep: true,
+    });
+
+    const savedObjects: string[] = await createPromiseFromStreams([
+      exportStream,
+      createMapStream((savedObject) => {
+        return JSON.stringify(savedObject);
+      }),
+      createConcatStream([]),
+    ]);
+
+    return response.ok({
+      body: savedObjects.join('\n'),
+      headers: {
+        'Content-Disposition': `attachment; filename="content_pack.ndjson"`,
+        'Content-Type': 'application/ndjson',
+      },
+    });
+  },
+});
+
+const uploadContentPacksRoute = createServerRoute({
+  endpoint: 'POST /api/streams/{name}/content_packs',
```

Comment:
i wonder if the targets stream(s) should be part of the body to enable importing to multiple streams at once.

Reply:
you can't do `/api/streams/content`, that's interpreted as `/api/streams/${id}`

Reply:
`/api/streams/content/_import` would be fine right ? `/api/streams/content` would also be (looks like exact match takes precedence over paths with variables from my limited testing) but would cause issues for classic streams named `content`

Reply:
Not too convinced here tbh - we are planning a _bulk API anyway for all kinds of changes, that would capture the use case of importing to multiple places at once.

So you have
* Dedicated API for simple cases, easily discoverable etc.
* _bulk for power users and automation

wdyt?

Reply:
I'm inclined to move forward with single target routes for now and revisit the bulk case when it's needed